packages/image/src/Image.svelte

```diff
@@ -0,0 +1,173 @@
+				[width, width * 2].map(
+					(w) => all_sizes.find((p) => p >= w) || all_sizes[all_sizes.length - 1]
+				)
+			)
+		];
+		return { widths, kind: 'x' };
+	}
+
+	/**
+	 * @param {string} src
+	 */
+	function matches_domain(src) {
+		const url = new URL(src, 'http://n'); // if src is protocol relative, use dummy domain
+		if (url.href === src) {
+			return domains.some((domain) => url.hostname === domain);
+		} else {
+			return true; // relative urls are always ok
+		}
+	}
+</script>
+
+<script>
+	/** @type {string | import('vite-imagetools').Img | import('vite-imagetools').Picture} */
+	export let src;
+	/** @type {string} */
+	export let alt;
+	/** @type {number}*/
+	export let width;
+	/** @type {number}*/
+	export let height;
+	/** @type {string | undefined} */
+	export let sizes = undefined;
+	/** @type {string | undefined} */
+	export let style = undefined;
+	/** @type {string} */
+	export let provider = 'default';
+	/** @type {any} */
+	export let providerOptions = undefined;
+	/** @type {'lazy' | 'eager' | undefined} */
+	export let loading = undefined;
+	/** @type {boolean} */
+	export let priority = false;
+
+	let srcset = '';
+	let _src = '';
+
+	$: {
+		if (typeof src !== 'string') {
+			srcset = (src.img || src).srcset?.map((i) => `${i.src} ${i.w}w`)?.join(', ');
+			_src = (src.img || src).src;
+			width = width || (src.img || src).w;
+			height = height || (src.img || src).h;
+		} else {
+			if (matches_domain(src)) {
+				const p = providers[provider];
+				if (DEV && !p) {
+					throw new Error(
+						`No provider named "${provider}" found. Available providers are ${Object.keys(
+							providers
+						).join(', ')}. Check the kit.images.providers configuration in your svelte.config.js.`
+					);
+				}
+
+				const widths = getWidths(width, !!style && /width: \d+(px|em|rem)/.test(style), sizes);
+				srcset = widths.widths
+					.map((w) => {
+						const url = DEV
+							? src
+							: p.getURL({
+									src: /** @type {string} */ (src),
+									width: w,
+									height: Math.round(w * (width / height)),
+									options: providerOptions
+							  });
+						return `${url} ${w}${widths.kind}`;
+					})
+					.join(', ');
+				_src = srcset.at(-1) ?? src;
+			} else {
+				srcset = '';
+				_src = src;
+			}
+		}
+	}
+</script>
+
+<svelte:head>
+	{#if priority}
+		<!-- don't set href because older browsers which don't support imagesrcset would probably load the wrong image -->
+		<link rel="preload" as="image" imagesrcset={srcset} imagesizes={sizes} fetchpriority="high" />
+	{/if}
+</svelte:head>
+
+<!-- Chrome with cache disabled seems to reload the src if it is reset during hydration which redownloads it -->
+{#if src.img}
+	<picture>
+		{#each Object.entries(src.sources) as [format, images]}
+			<source srcset={images.map((i) => `${i.src} ${i.w}w`).join(', ')} type={'image/' + format} />
+		{/each}
+		<img
+			{...$$restProps}
```

Comment:
do we need restprops here or is it possible to just forward some known props? output with restprops is a bit dfferent.